assets/blocks/quiz/quiz-block/use-add-existing-questions.js

```diff
@@ -0,0 +1,66 @@
+/**
+ * WordPress dependencies
+ */
+import { select, useDispatch } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import { createQuestionBlock, findQuestionBlock } from '../data';
+
+const API_PATH = '/sensei-internal/v1/question-options';
+
+/**
+ * Add existing questions to the quiz block.
+ *
+ * @param {string} clientId The quiz block client id.
+ * @return {Function} Function that takes an array of question IDs and returns a Promise.
+ */
+export const useAddExistingQuestions = ( clientId ) => {
+	const questionBlocks = select( 'core/block-editor' ).getBlocks( clientId );
```

Comment:
I would filter the blocks which have a title directly to avoid the extra complexity later.

Reply:
Do you mean in `findQuestionBlock`? We already filter there (that is a shared method).

Reply:
Nope, I mean instead of getting all questions we could remove the ones that don't have a title. Then we don't need to calculate the index based on that and we can simply add it in the end.

Reply:
I think we only would do that to the last question (the one from `useAutoInserter`). We wouldn't want to filter out all the empty title questions. 

For example, using 0-index:
0) Question with title (A)
1) [empty]
2) [empty]
3) Question with title (B)
4) [empty, from `useAutoInserter`]

We'd want the new questions to start getting inserted at index 4. If we filtered out all empty questions here, it would get inserted before `Question with title (B)` when calculating the index.

Reply:
Aaah cool! I thought that empty questions can be only in the end for some reason!